executor/analyze.go

```diff
@@ -515,3 +526,83 @@ func (e *AnalyzeFastExec) buildStats() (hists []*statistics.Histogram, cms []*st
 	// TODO: do fast analyze.
 	return nil, nil, nil
 }
+
+type analyzeIndexIncrementalExec struct {
+	AnalyzeIndexExec
+	index *statistics.Index
+}
+
+func analyzeIndexIncremental(idxExec *analyzeIndexIncrementalExec) statistics.AnalyzeResult {
+	idx := idxExec.index
+	highBound := idx.Histogram.GetUpper(idx.Len() - 1)
+	values, err := codec.Decode(highBound.GetBytes(), len(idxExec.idxInfo.Columns))
+	if err != nil {
+		return statistics.AnalyzeResult{Err: err}
+	}
+	ran := ranger.Range{LowVal: values, HighVal: []types.Datum{types.MaxValueDatum()}}
+	hist, cms, err := idxExec.buildStats([]*ranger.Range{&ran}, false)
+	if err != nil {
+		return statistics.AnalyzeResult{Err: err}
+	}
+	oldHist, oldCMS, err := idx.RemoveUpperBound(idxExec.ctx.GetSessionVars().StmtCtx, values)
```

Comment:
Why do we have to remove upper bound after request instead of excluding low value in request range like `analyzePKIncremental`?

Reply:
Because the index may not be unique, and there may be newly added rows that are the same as the upper bound.